tools/stringparser/main.c

```diff
@@ -59,7 +59,10 @@ int main(int argc, char* argv[]) {
     if (strcmp("-", *astr->sval) == 0) {
         from_stdin = 1;
         char *tmp_str = malloc(512);
-        fgets(tmp_str, 512, stdin);
+        if (fgets(tmp_str, 512, stdin) == NULL) {
+            printf("fgets(512, stdin) failed\n");
```

Comment:
This should really use the PERROR macro instead

Reply:
Yeah, I'll make a proper fix instead ..